### PR TITLE
add test for ReserveFund.sol

### PR DESCRIPTION
**Pull request by alikonuk1**

### CODE REVIEW

File: src/test/12_ReserveFund.t.sol [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+        vm.startPrank(liquidatorAddress);
+        reserveFund.withdraw(50e18, address(stableCoin), address(liquidatorAddress));
+        vm.stopPrank();
+        require(stableCoin.balanceOf(address(liquidatorAddress)) == 50e18);
+    }
+
+    function testFailOwnerOrLiquidatorWithdraw() public {
+        vm.startPrank(randomAddress);
+        reserveFund.withdraw(50e18, address(stableCoin), address(randomAddress));
```
Comment (j-vp) [inline src/test/12_ReserveFund.t.sol:62]: Preferably a vm.expectRevert is used to check more explicit on the revert reasons.

File: src/test/12_ReserveFund.t.sol [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        reserveFund.setLiquidator(address(liquidatorAddress));
+        vm.stopPrank();
+
+        vm.startPrank(liquidatorAddress);
+        reserveFund.withdraw(50e18, address(stableCoin), address(liquidatorAddress));
+        vm.stopPrank();
+        require(stableCoin.balanceOf(address(liquidatorAddress)) == 50e18);
+    }
+
+    function testFailOwnerOrLiquidatorWithdraw() public {
```
Comment (j-vp) [inline src/test/12_ReserveFund.t.sol:60]: Preferably a vm.expectRevert is used to check more explicit on the revert reasons.

File: src/test/12_ReserveFund.t.sol [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        stableCoin = new ERC20Mock("Test Stable Coin", "USD", 18);
+        stableCoin.mint(address(reserveFund), 100e18);
+        vm.stopPrank();
+    }
+
+    function testOwnerSetLiquidator() public {
+        vm.startPrank(ownerAddress);
+        reserveFund.setLiquidator(address(liquidatorAddress));
+        vm.stopPrank();
+    }
```
Comment (j-vp) [inline src/test/12_ReserveFund.t.sol:35]: Add vm.assert for liquidator owner.

File: src/test/12_ReserveFund.t.sol [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    function testFailUserSetLiquidator() public {
```
Comment (j-vp) [inline src/test/12_ReserveFund.t.sol:37]: Preferably a vm.expectRevert is used to check more explicit on the revert reasons.

**Comment by alikonuk1** [comment]
Projects foundry version is lower than the one i use on my end, so somehow vm.expectRevert does not work as intended because of the version differences, but i added anyways and it should pass on the lower version. Also trying to make it work again on new version too.

**Comment by alikonuk1** [comment]
vm.expectRevert now working and tests are passing.

**Review by j-vp — APPROVED** [review]
Gtg!

**Pull request merged**